src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthTests.cs

```diff
@@ -0,0 +1,537 @@
+        public async Task SrcAndDstAsLiteralArrays()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        Buffer.BlockCopy(new int[] {1, 2, 3, 4}, 0, new int[] {0, 0, 0, 0}, 0, 8);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Buffer.Blockcopy(new Integer() {1, 2, 3, 4}, 0, new Integer() {0, 0, 0, 0}, 0, 8)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NamedArgumentsNotInOrder()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+        
+        Buffer.BlockCopy(srcOffset: 0, src: src, count: [|src.Length|], dstOffset: 0, dst: dst);
+    }
+}
+");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(srcOffset:=0, src:=src, count:=[|src.Length|], dstOffset:=0, dst:=dst)
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task NonLocalArrays()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] src = new int[] {1, 2, 3, 4};
+        int[] dst = new int[] {0, 0, 0, 0};
+        
+        SomeFunction(src, dst);
+    }
+
+    static void SomeFunction(int[] src, int[] dst)
+    {
+        Buffer.BlockCopy(src, 0, dst, 0, [|src.Length|]);
+    }
+}
+");
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim src = New Integer() {1, 2, 3, 4}
+        Dim dst = New Integer() {0, 0, 0, 0}
+
+        SomeFunction(src, dst)
+    End Sub
+
+    Sub SomeFunction(ByRef src As Integer(), ByRef dst As Integer())
+        Buffer.BlockCopy(src, 0, dst, 0, [|src.Length|])
+    End Sub
+End Module
+");
+        }
+
+        [Fact]
+        public async Task SrcArrayAsFunctionReturn()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        int[] dst = new int[] {0, 0, 0, 0};
+        
+        Buffer.BlockCopy(GetSrcArray(), 0, dst, 0, [|dst.Length|]);
+    }
+
+    static int[] GetSrcArray()
+    {
+        return new int[] { 1, 2, 3, 4 };
+    }
+}
+");
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+
+Module Program
+    Sub Main(args As String())
+        Dim dst = New Integer() {0, 0, 0, 0}
+
+        Buffer.BlockCopy(GetSrcArray(), 0, dst, 0, [|dst.Length|])
+    End Sub
+
+    Function GetSrcArray()
+        Return New Integer() {1, 2, 3, 4}
+    End Function
+End Module
+
+");
+        }
+    }
```

Comment:
Please add a test with a global array
```cs
class Program
{
    private char[] arr1;
    private char[] arr2;

    public void Test()
    {
        Buffer.BlockCopy(arr1, 0, arr2, 0, arr1.Length);
    }
}
```

Reply:
and another test with Array type:
```cs
Array arr1 = Array.CreateInstance(typeof(byte), 2), arr2 = Array.CreateInstance(typeof(byte), 2);
Buffer.BlockCopy(arr1, 0, arr2, 0, arr1.Length);
```

Reply:
Let me know if you need help with these. AFAICT these are the last 2 comments needing to be addressed.